torch/distributed/rpc/api.py

```diff
@@ -197,14 +197,8 @@ def _all_gather(obj):
             )
             worker_name_to_response_future_dict[follower_name] = fut
         for follower_name, fut in worker_name_to_response_future_dict.items():
-            try:
-                fut.wait()
-            except RuntimeError as ex:
-                logger.error(
-                    "{worker_name} failed to respond to 'Shutdown Proceed.' request in {timeout}".format(
-                        worker_name=follower_name, timeout=timeout
-                    )
-                )
+            fut.wait()
```

Comment:
Previously we would call `wait()` on all futures during shutdown, but now we throw on the first one and probably don't wait for the other futures.

It would be nice to keep the same behavior as before. Can we add a try-catch here and save the exception? We can probably only throw the first/last exception we see and catch that in `_wait_all_workers`.